### ISSUE TITLE
How to set initial data?

### ISSUE THREAD

**Issue by hvanoch** [neighborhood999/vue-signature-pad#191]
It looks like there is no way to prefill the component with a previous data url?

**Comment by neighborhood999** [comment]
See `fromData` or `fromDataURL` methods:

- https://github.com/neighborhood999/vue-signature-pad#methods
- https://github.com/szimek/signature_pad#api

**Comment by hvanoch** [comment]
Yes I know of those methods. But if you use v-if on the component and you make the component "visible" you can not reference it with `this.$refs.signature_path` immediately after,  because it is not loaded yet. 
I found a little hack for this problem, by using 

> this.$nextTick(() => { 
>    this.$refs.signature_pad.fromDataURL(this.signature)
>})

Having a property available on component that contains the initial signature seems like a lot more developer friendly

**Comment by neighborhood999** [comment]
Can you create this example by codesandbox? This is good for me to think this new API.

**Comment by hvanoch** [comment]
https://codesandbox.io/s/vue-signature-pad-demo-0tg6g

here is an example of the issue that I have.

There are other solutions that would work to. Like emit an event when the component is loaded and destroyed. Then we can listen to those events in the parent component.

**Comment by neighborhood999** [comment]
Hi @hvanoch, thank you for providing this case! I think this issue is liked #187, am I right?

I have an idea for fixing this, but I'm not sure other peoples use cases.

`VueSignaturePad` can provide `visible` API as props, user can control visible by their parent component.

then we can destructuring this prop and check if it needs to be rendered:

https://github.com/neighborhood999/vue-signature-pad/blob/17e37385a32ccb5f8bf47b594ab4adc538748c65/src/components/VueSignaturePad.js#L173-L183

e.g:

```js
const { width, height, customStyle, visible } = this;

return createElement(
  'div',
  {
	style: {
	  width,
	  height,
	  display: visible ? 'block' : 'none',
	  ...customStyle
  	}
  },
)
```

cc @Leemanbok @scp-nm @alexsante @pynner

**Comment by hvanoch** [comment]
That still doesn't really solve the issue for setting the initial data when the component loads. I think the easiest and the most flexible will be to fire events when components is loaded and destroyed.
https://vuejs.org/v2/guide/components-custom-events.html

**Comment by KAi2020RM** [comment]
Hello to all,

I also have the problem to initialize existing data at the beginning. I use the [SignaturePad within a stepper](https://codesandbox.io/s/tender-fog-1se94) (example Step 2) in Vuetify. 

The first problem is that I am forced to call the onBegin method explicitly at startup via the options, otherwise the SignaturePad does not work. Height and width of the canvas is 0. 

The second problem is that I would like to initialize data at the beginning. This also only works if I click twice on the SignaturePad (height and width of the canvas is 0 at the beginning). But the goal is that in this example, the letter "K" is displayed directly in the Signature Pad. 

So I have made an example (SignaturePad in Step 2) (Quick & Dirty), which you can find here [https://codesandbox.io/s/tender-fog-1se94](https://codesandbox.io/s/tender-fog-1se94), so that you can see directly what my problem is.

It would really help me a lot if someone had an idea how I could integrate the signature directly into the SignaturePad so that the signature is displayed directly when you call up the second step!

Thanks for your support!

**Comment by benedict-w** [comment]
@KAi2020RM I have a similar issue using fromDataURL to load an image when the canvas is hidden - has anyone resolved this?

**Comment by debaetsr** [comment]
I solved this with a timer, see https://www.npmjs.com/package/vue-timers

After toggling visibility, I start a timer for 500ms, after which the $refs is available.

**Comment by rezelute** [comment]
Has this project been abandoned or will this fix be included at any point?
